internal/controller/nats/controller.go

```diff
@@ -25,26 +25,36 @@ import (
 	"github.com/kyma-project/nats-manager/pkg/k8s/chart"
 	"github.com/kyma-project/nats-manager/pkg/manager"
 	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
 )
 
 const (
-	NATSFinalizerName = "nats.operator.kyma-project.io/finalizer"
+	NATSFinalizerName   = "nats.operator.kyma-project.io/finalizer"
+	ControllerName      = "nats-manager"
+	ManagedByLabelKey   = "app.kubernetes.io/managed-by"
+	ManagedByLabelValue = ControllerName
 )
 
 // Reconciler reconciles a Nats object.
+//
+//go:generate mockery --name=Controller --dir=../../../vendor/sigs.k8s.io/controller-runtime/pkg/controller --outpkg=mocks --case=underscore
 type Reconciler struct {
 	client.Client
-	kubeClient    k8s.Client
-	chartRenderer chart.Renderer
-	Scheme        *runtime.Scheme
-	recorder      record.EventRecorder
-	logger        *zap.SugaredLogger
-	NATSManager   manager.Manager
+	controller.Controller
+	kubeClient                  k8s.Client
+	chartRenderer               chart.Renderer
+	Scheme                      *runtime.Scheme
+	recorder                    record.EventRecorder
+	logger                      *zap.SugaredLogger
+	NATSManager                 manager.Manager
+	destinationRuleWatchStarted bool
```

Comment:
Why the following properties are exported?

client.Client
controller.Controller
Scheme
NATSManager

Reply:
I have update the following fields:
```
controller.Controller
Scheme
NATSManager
```

The `client.Client` is still exported so can refer to its methods from reconciler e.g `r.Create()`.